src/Symfony/Component/HttpKernel/EventListener/ErrorListener.php

```diff
@@ -65,18 +76,25 @@ public function onKernelException(ExceptionEvent $event)
 
             $prev = $e;
             do {
-                if ($exception === $wrapper = $prev) {
+                if ($throwable === $wrapper = $prev) {
                     throw $e;
                 }
             } while ($prev = $wrapper->getPrevious());
 
             $prev = new \ReflectionProperty($wrapper instanceof \Exception ? \Exception::class : \Error::class, 'previous');
             $prev->setAccessible(true);
-            $prev->setValue($wrapper, $exception);
+            $prev->setValue($wrapper, $throwable);
 
             throw $e;
         }
 
+        foreach ($this->exceptionsMapping as $exception => $config) {
+            if ($throwable instanceof $exception && $config['status_code']) {
+                $response->setStatusCode($config['status_code']);
```

Comment:
does this work? The response is rendered L71 before this happens so whilst the response will have the correct status code the page rendered may not be matching. For example the ErrorController will render a page for a 500 although the status code will be altered here

Reply:
I don't undestant. The status code you write in the configuration is the one you want. 
It is not used to "match" something. It overrides the current value.

Reply:
_It overrides the current value_

Of the created response.

In the case you are dealing with an API, you probably do not care. However, if you are dealing with an HTML response, then you probably have a different template if it's a 400 or a 500. From what I see here although you override the status code from a 500 to a 400 for example, the rendered page would still look like you had a 500 (and with the default template it shows the status code, i.e. 500).

Reply:
We might need to update `TwigErrorRenderer` to take this new exceptions mapping into account.